File: unique_problems/two_sum2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        low = 0
        high = len(numbers)-1
        while(low<high):
            tsum = numbers[low]+numbers[high]
            if(target==tsum):
                return [low+1,high+1]
            elif(target>tsum):
                low+=1
            else:
                high-=1
        # hashmap = {}
        # for i in range(len(numbers)):
        #     ele = numbers[i]
        #     if target-ele in hashmap:
        #         return sorted([i+1,hashmap[target-ele]+1])
        #     else:
        #         hashmap[ele] = i

File: unique_problems/two_sum2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int low = 0;
        int high = numbers.size()-1;
        while(low<high){
            int tsum = numbers[low]+numbers[high];
            if(tsum==target){
                return {low+1,high+1};
            }
            else if(target>tsum){
                low++;
            }
            else{
                high--;
            }
        }
        return {-1,-1};
    }
};